feat(sim): add SIH SITL build target and restructure sim docs (#26672)
PX4/PX4-Autopilot: docs/en/dev_setup/building_px4.md

- **VENDOR:** The manufacturer of the board: `px4`, `aerotenna`, `airmind`, `atlflight`, `auav`, `beaglebone`, `intel`, `nxp`, etc.
  The vendor name for Pixhawk series boards is `px4`.
- **MODEL:** The _board model_ "model": `sitl`, `fmu-v2`, `fmu-v3`, `fmu-v4`, `fmu-v5`, `navio2`, etc.

> - **VARIANT:** Indicates particular configurations: e.g. `bootloader`, `cyphal`, `sih`, which contain components that are not present in the `default` configuration.

Reviewer: ```suggestion
- **VARIANT:** Indicates particular configurations: e.g. `bootloader`, `cyphal`, `sih`, which add or remove components to/from the `default` configuration.
```
